### PR TITLE
Move to HDP provided spark2 package

### PR DESCRIPTION
**Pull request by amithkanand**
We still need a patched jar with fixes for:
- [OOZIE-2606](https://issues.apache.org/jira/browse/OOZIE-2606)
- [OOZIE-2802](https://issues.apache.org/jira/browse/OOZIE-2802)

Also, `oozie-spark` action is available as technical preview with `HDP` distribution: 
```
In HDP 2.6, Oozie works with either Spark 1 or Spark 2 (not side-by-side deployments), 
but Spark 2 support for Oozie Spark action is available as a technical preview; it is not 
ready for production deployment. Configuration is through manual steps (not Ambari).
```

### CODE REVIEW

File: cookbooks/bach_spark/recipes/default.rb [hunk 1]
```diff
@@ -54,8 +64,11 @@
   helpers(Spark::Configuration)
 end
 
-link "/#{spark_bin_dir}/yarn/spark-yarn-shuffle.jar" do
-  to "#{spark_bin_dir}/yarn/spark-#{spark_pkg_version}-yarn-shuffle.jar"
+# For backward compatibility
+
+directory '/usr/spark' do
+   action :create
+   mode 0755
```
Comment (vt0r) [inline cookbooks/bach_spark/recipes/default.rb]: Style: `0o0755`

File: cookbooks/bcpc-hadoop/recipes/yarn_env.rb [hunk 2]
```diff
@@ -9,8 +9,9 @@
 if node.run_list.expand(node.chef_environment)
        .recipes.include?('bach_spark::default')
   yarn_env_generated_values[:YARN_USER_CLASSPATH] =
-    '/usr/spark/current/yarn/spark-yarn-shuffle.jar'
+    "/usr/hdp/current/spark2-client/aux/spark-#{node[:bcpc][:hadoop][:distribution][:active_release]}-yarn-shuffle.jar"
```
Comment (vt0r) [inline cookbooks/bcpc-hadoop/recipes/yarn_env.rb]: Style: Symbols --> Strings ([FC001](http://www.foodcritic.io/#FC001))

Could also use the bin dir attribute here.

File: cookbooks/bach_spark/attributes/default.rb [hunk 3]
```diff
@@ -1,60 +1,68 @@
...
-default[:spark][:download][:url] = 'http://d3kbcqa49mib13.cloudfront.net'
-default[:spark][:download][:file][:name] = 'spark-2.1.0-bin-hadoop2.7'
-default[:spark][:download][:file][:type] = 'tgz'
-default[:spark][:download][:dir] = "/home/#{user}/chef-bcpc/bins"
-default[:spark][:package][:install_meta] = false
-default[:spark][:package][:base] = '/usr/spark'
-default[:spark][:package][:prefix] = 'spark'
-default[:spark][:package][:version] = '2.1.0'
 default[:spark][:hdfs_url] = node['bcpc']['hadoop']['hdfs_url']
-default[:spark][:bin][:dir] = "#{node[:spark][:package][:base]}/"\
```
Comment (vt0r) [inline cookbooks/bach_spark/attributes/default.rb]: Should probably leave this attribute here, but just make the value `'/usr/hdp/current/spark2-client'`

File: cookbooks/bach_spark/recipes/cluster_install.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # vim: tabstop=2:shiftwidth=2:softtabstop=2
 
-spark_pkg_version = node[:spark][:package][:version]
-spark_bin_dir = node[:spark][:bin][:dir]
+spark_pkg_version = node['bcpc']['hadoop']['distribution']['active_release']
+spark_bin_dir = '/usr/hdp/current/spark2-client'
```
Comment (vt0r) [inline cookbooks/bach_spark/recipes/cluster_install.rb]: Should return to using the attribute here.

File: cookbooks/bach_spark/recipes/default.rb [hunk 5]
```diff
@@ -23,20 +23,30 @@
 # the fpm invocation that creates the debian package.
 #
 
-spark_pkg_version = node[:spark][:package][:version]
-spark_bin_dir = node[:spark][:bin][:dir]
+spark_bin_dir = '/usr/hdp/current/spark2-client'
```
Comment (vt0r) [inline cookbooks/bach_spark/recipes/default.rb]: Also return to the attribute here, please.

File: cookbooks/bach_spark/recipes/default.rb [hunk 6]
```diff
@@ -23,20 +23,30 @@
...
-    action :upgrade
-  end
-else
-  package "spark-#{spark_pkg_version}" do
-    action :install
-  end
+directory "/etc/spark2/conf.#{node.chef_environment}" do
+  owner "root"
+  group "root"
+  mode 00755
```
Comment (vt0r) [inline cookbooks/bach_spark/recipes/default.rb]: Style: Single quotes around user/group + octal literal for mode

File: cookbooks/bach_spark/recipes/default.rb [hunk 7]
```diff
@@ -45,7 +55,7 @@
   helpers(Spark::Configuration)
 end
 
-template "#{spark_bin_dir}/conf/spark-defaults.conf" do
+template "/etc/spark2/conf/spark-defaults.conf" do
```
Comment (vt0r) [inline cookbooks/bach_spark/recipes/default.rb]: Looks like `/etc/spark2/conf` could be a new attribute (something like `node['spark']['conf']['dir']`)?

**Comment by amithkanand** [comment]
Implemented requested changes!

**Pull request merged**